packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -66,6 +67,7 @@ const deploymentInProgressErrorMessage = (context: $TSContext) => {
 };
 
 export async function run(context: $TSContext, resourceDefinition: $TSObject) {
+  context = await ensureRootStack(context);
```

Comment:
We should solve this differently and not mutate the incoming parameter here, ideally that should have a 'readonly' modifier

Reply:
The problem is the existing stack initialization code mutates context so therefore the ensureRootStack function also does (because it calls into the existing code if not initialized). ensureRootStack doesn't need to return context and it doesn't need to be reassigned here but I did it that way to make it more obvious that this function is mutating the context